osu.Game/Online/Multiplayer/MultiplayerClient.cs

```diff
@@ -96,6 +96,9 @@ public override Task ChangeState(MultiplayerUserState newState)
             if (!IsConnected.Value)
                 return Task.CompletedTask;
 
+            if (newState == MultiplayerUserState.Spectating)
+                return Task.CompletedTask; // Not supported yet.
```

Comment:
I guess this is here to avoid an exception returning from the server?

Reply:
Correct